### PR TITLE
JitArm64: Initial implementation of the BLR optimization.

### PR DESCRIPTION
**Pull request by degasus**
This PR lacks the guard page to catch stack overflows. Else it seems to be fully functional. There haven'T been much performance optimizations, this PR tries to upload readable code. Still *much* faster than master.

### CODE REVIEW

File: Source/Core/Core/PowerPC/JitArm64/Jit.cpp [hunk 1]
```diff
@@ -192,6 +193,143 @@ void JitArm64::DoDownCount()
   gpr.Unlock(WA, WB);
 }
 
+void JitArm64::ResetStack()
+{
+  LDR(INDEX_UNSIGNED, X0, PPC_REG, PPCSTATE_OFF(stored_stack_pointer));
+  SUB(SP, X0, 16);
+}
+
+void JitArm64::WriteLKExit(u32 destination, u32 after)
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/JitArm64/Jit.cpp]: `after` should probably be `exit_address_after_return` or something more indicative to what it is. Ditto for the two other functions below that also have `u32 after`

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_Branch.cpp [hunk 2]
```diff
@@ -211,7 +232,15 @@ void JitArm64::bcctrx(UGeckoInstruction inst)
 
   LDR(INDEX_UNSIGNED, WA, PPC_REG, PPCSTATE_OFF(spr[SPR_CTR]));
   AND(WA, WA, 30, 29);  // Wipe the bottom 2 bits.
-  WriteExit(WA);
+
+  if (inst.LK_3)
+  {
+    WriteLKExit(WA, js.compilerPC + 4);
```
Comment (phire) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_Branch.cpp]: Unrelated to the main part of this PR, but constant propagation might have resolved the jump address to a constant.

**Comment by eddiebrook** [comment]
some games crash in game after few seconds of gameplay. the only way to fix this is disabling multicore.

for example dbz tenkiachi 3 wii

**Comment by degasus** [comment]
@eddiebrook Just to be sure, you haven't disabled "SyncOnSkipIdle" in the Dolphin.ini, have you?

It's quite likely that I've broken something here, but I doubt single core may fix it.

**Comment by eddiebrook** [comment]
No
i have SyncOnSkipIdle = True

**Comment by eddiebrook** [comment]
for example in the game related before.

1 i need to quicksave before  enter to fight (or crash)
2 back to dolphin menú and disable multicore
3 quickload the game again, enter in battle (no crash)
4 quicksave again and back to dolphin menú
5 enabled dualcore again and quickload the game
6 the game work fine until enter to another scenario/fight

**Comment by degasus** [comment]
May you try if booting in single core also crashes? Loading and storing a savestate touches the JIT state, so it might be savestate related instead of dual core.

**Comment by brujo5** [comment]
I tried that to and working fine in single core. (Loaded Normal save)

**Comment by degasus** [comment]
@eddiebrook PR #4983 fixes a random crash which sounds like the one you're reporting here. Does this PR fixes the regressions of this one?

**Pull request merged**